src/components/notifications/notification_item.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import TimeAgo from 'react-timeago';
+import styled, { keyframes } from 'styled-components';
+
+import { themeColors, themeDimensions } from '../../util/theme';
+import { tokenAmountInUnits } from '../../util/tokens';
+import { Notification, OrderSide } from '../../util/types';
+import { NotificationCancelIcon } from '../common/icons/notification_cancel_icon';
+import { NotificationCheckmarkIcon } from '../common/icons/notification_checkmark_icon';
+import { NotificationProcessingIcon } from '../common/icons/notification_processing_icon';
+
+interface Props {
+    item: Notification;
+}
+
+interface State {
+    active: boolean;
+    pending: boolean;
+}
+
+const NotificationWrapper = styled.div<{ active?: boolean }>`
+    align-items: center;
+    background-color: ${props => (props.active ? themeColors.rowActive : 'transparent')};
+    border-bottom: 1px solid ${themeColors.borderColor};
+    display: flex;
+    justify-content: space-between;
+    padding: 20px ${themeDimensions.horizontalPadding};
+
+    &:last-child {
+        border-bottom: none;
+    }
+`;
+
+const NotificationContent = styled.div`
+    flex-grow: 1;
+    padding-right: 25px;
+`;
+
+const NotificationTitle = styled.h2`
+    color: #000;
+    font-size: 16px;
+    font-weight: 600;
+    line-height: 1.2;
+    margin: 0 0 8px;
+`;
+
+const NotificationText = styled.p`
+    color: ${themeColors.textLight};
+    font-size: 16px;
+    font-weight: 400;
+    line-height: 1.2;
+    margin: 0;
+`;
+
+const rotate = keyframes`
+  from {
+    transform: rotate(0deg);
+  }
+
+  to {
+    transform: rotate(360deg);
+  }
+`;
+
+const NotificationIconSpin = styled.div`
+    animation: ${rotate} 1.5s linear infinite;
+`;
+
+const NotificationIcon = styled.div`
+    flex-shrink: 0;
+`;
+
+class NotificationItem extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            active: false,
+            pending: false,
+        };
+    }
+
+    public componentDidMount = () => {
+        const { item } = this.props;
+
+        if (item.kind === 'MarketNotification') {
+            this.setState({
+                active: true,
+                pending: true,
+            });
+
+            /* tslint:disable-next-line:no-floating-promises */
```

Comment:
I believe this should do the trick:
```typescript
try {
    await item.tx;
} finally {
    this.setState({ pending: false });
}
```
Source: https://dev.to/annarankin/finally-in-promises--trycatch-2c44 .

Reply:
I left it as it was but added the `await`, that did the trick (I prefer it that way over the `try/catch`, since IMO the intention is clearer, but YMMV).